src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -5254,35 +5254,14 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
               // implicit scopes which are accessible via lhs can be candidates for satisfying
               // implicit (conversions to) arguments of name.
 
-              // We have to introduce the ValDef before its use here, so we walk up the
-              // context tree and attach it to the original root of this expression. It will
-              // be extracted and inserted by insertStabilizer when typer unwinds out of this
-              // expression. Stabilized args are introduced in the block arg expression.
-              val insertionContext = context.nextEnclosing { ctx =>
-                def isInsertionNode(tree: Tree) =
-                  tree match {
-                    case _: Apply | _: TypeApply | _: Select => true
-                    case _                                   => false
-                  }
-                def isEnclosingInsertionNode(tree: Tree) =
-                  tree match {
-                    case Apply(_, args)           => args.contains(ctx.tree)
-                    case _: TypeApply | _: Select => false
-                    case _                        => true
-                  }
-                isInsertionNode(ctx.tree) && isEnclosingInsertionNode(ctx.outer.tree)
-              }
-
-              if (insertionContext != NoContext) {
-                val vsym = insertionContext.owner.newValue(freshTermName(nme.STABILIZER_PREFIX), qual.pos.focus, SYNTHETIC | ARTIFACT | STABLE)
-                vsym.setInfo(uncheckedBounds(qual.tpe))
-                insertionContext.scope enter vsym
-                val vdef = atPos(vsym.pos)(ValDef(vsym, focusInPlace(qual)) setType NoType)
-                qual.changeOwner(insertionContext.owner -> vsym)
-                addStabilizingDefinition(insertionContext.tree, vdef)
-                val newQual = Ident(vsym) setType singleType(NoPrefix, vsym) setPos qual.pos.focus
-                return typedSelect(tree, newQual, name)
-              }
+              val vsym = context.owner.newValue(freshTermName(nme.STABILIZER_PREFIX), qual.pos.focus, SYNTHETIC | ARTIFACT | STABLE)
+              vsym.setInfo(uncheckedBounds(qual.tpe))
+              val vdef = atPos(vsym.pos)(ValDef(vsym, focusInPlace(qual)) setType NoType)
+              unit.pendingStabilizers ::= vdef
+              qual.changeOwner(context.owner -> vsym)
+              tree.updateAttachment(StabilizedQualifier(vsym))
```

Comment:
This is too brittle. If the qualifier is changed (implicit conversion to a type that has the selected member), the `tree` here is not the same as the `original` below in `definesStabilizer`, so the attachment is not found and the stabilizer is not added. (That's what breaks compiling `FastTrack.scala`). Need to find a better backdoor.

Reply:
I can exercise it before valentine's day is over. If you have any other ideas. Isn't your day over? Yes, but your night is just getting started.

Reply:
Pushed another attempt; I promise not to touch it for the rest of the weekend.